File: v1/ecmascript/rna-transcription/rna-transcription.js
export default class Transcriptor {
    toRna(dna) {
        return Array.prototype.map.call(dna, this.dnaToRna).join('');
    }

    dnaToRna(nucleotide) {
        switch (nucleotide) {
            case 'G': return 'C';
            case 'C': return 'G';
            case 'T': return 'A';
            case 'A': return 'U';
        }
    }
}

File: swift/rna-transcription/Sources/RnaTranscription/RnaTranscription.swift
func toRna(_ dna: String) -> String {
  String(dna.map(toComplement))
}

func toComplement(_ nucleotide: Character) -> Character {
  switch (nucleotide) {
    case "G": return "C"
    case "C": return "G"
    case "T": return "A"
    case "A": return "U"
    default: return nucleotide
  }
}
